connector/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/MLUtils.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.ml
+
+import scala.reflect.ClassTag
+
+import org.apache.spark.connect.proto
+import org.apache.spark.ml.param.{ParamMap, Params}
+import org.apache.spark.sql.{DataFrame, Dataset}
+import org.apache.spark.sql.connect.planner.SparkConnectPlanner
+import org.apache.spark.sql.connect.service.SessionHolder
+
+object MLUtils {
+
+  def setInstanceParams(instance: Params, paramsProto: proto.Params): Unit = {
+    import scala.collection.JavaConverters._
+    paramsProto.getParamsMap.asScala.foreach { case (paramName, paramValueProto) =>
+      val paramDef = instance.getParam(paramName)
+      val paramValue = parseParamValue(paramDef.paramValueClassTag.runtimeClass, paramValueProto)
+      instance.set(paramDef, paramValue)
+    }
+    paramsProto.getDefaultParamsMap.asScala.foreach { case (paramName, paramValueProto) =>
+      val paramDef = instance.getParam(paramName)
+      val paramValue = parseParamValue(paramDef.paramValueClassTag.runtimeClass, paramValueProto)
+      instance._setDefault(paramDef -> paramValue)
+    }
+  }
+
+  def parseParamValue(paramType: Class[_], paramValueProto: proto.Expression.Literal): Any = {
```

Comment:
maybe we don't need the `paramType`? The Literal already carries the type.

I guess we can use `LiteralValueProtoConverter.{toCatalystValue, toArrayData}` here

Reply:
No. The purpose of `paramType` is:
in pyspark, we cannot distinguish int/long, double/float, so when send it to server, int becomes long, and float becomes double, we need to get the accurate param type otherwise JVM will raise error in runtime.

Reply:
for int/long case, we can specify the DataType in Python Client

https://github.com/apache/spark/blob/c99a632fea74136964b27b28563115fe2d7667b3/python/pyspark/sql/connect/expressions.py#L165-L171

Reply:
> for int/long case, we can specify the DataType in Python Client

This does not help.
Because the pyspark estimator Params it does not record the type attribute, so we cannot distinguish int/long, double/float for python side param values.

Reply:
I see, we don't have `IntParam` in PySpark